### PR TITLE
CodePipeline - implement CRUD endpoints

### PR DESCRIPTION
**Pull request by gruebel**
I also updated the implementation coverage and due to the reInvent there are a lot of changes.

I actually wanted to implement the the tag endpoints, but I think it is better to do it in a second PR :)

### CODE REVIEW

**Comment by gruebel** [comment]
@mikegrima I closed my old PR #2640, because I somehow messed up the rebasing :(

File: moto/codepipeline/models.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+from moto.iam import iam_backends
+
+from moto.codepipeline.exceptions import (
+    InvalidStructureException,
+    PipelineNotFoundException,
+    ResourceNotFoundException,
+)
+from moto.core import BaseBackend, BaseModel
+
+DEFAULT_ACCOUNT_ID = "123456789012"
```
Comment (mikegrima) [inline moto/codepipeline/models.py]: Can you pull this from `moto.core`? A recent PR added this.

Comment (gruebel) [reply moto/codepipeline/models.py]: Done :)

**Comment by mikegrima** [comment]
Will merge when tests pass.

**Comment by gruebel** [comment]
> Will merge when tests pass.

Ready to go!

**Pull request merged**